src/mat/4C_mat_fourieraniso.cpp

```diff
@@ -0,0 +1,131 @@
+// This file is part of 4C multiphysics licensed under the
+// GNU Lesser General Public License v3.0 or later.
+//
+// See the LICENSE.md file in the top-level for license information.
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include "4C_mat_fourieraniso.hpp"
+
+#include "4C_comm_pack_helpers.hpp"
+#include "4C_global_data.hpp"
+#include "4C_mat_par_bundle.hpp"
+
+FOUR_C_NAMESPACE_OPEN
+
+/*----------------------------------------------------------------------*
+ *----------------------------------------------------------------------*/
+Mat::PAR::FourierAniso::FourierAniso(const Core::Mat::PAR::Parameter::Data& matdata)
+    : Parameter(matdata),
+      // be careful: capa_ := rho * C_V, e.g contains the density
+      capa_(matdata.parameters.get<double>("CAPA")),
+      conduct_(matdata.parameters.get<std::vector<double>>("CONDUCT"))
+{
+}
+
+/*----------------------------------------------------------------------*
+ *----------------------------------------------------------------------*/
+std::shared_ptr<Core::Mat::Material> Mat::PAR::FourierAniso::create_material()
+{
+  return std::make_shared<Mat::FourierAniso>(this);
+}
+
+/*----------------------------------------------------------------------*
+ *----------------------------------------------------------------------*/
+Mat::FourierAnisoType Mat::FourierAnisoType::instance_;
+
+/*----------------------------------------------------------------------*
+ *----------------------------------------------------------------------*/
+Core::Communication::ParObject* Mat::FourierAnisoType::create(
+    Core::Communication::UnpackBuffer& buffer)
+{
+  auto* fourieraniso = new Mat::FourierAniso();
+  fourieraniso->unpack(buffer);
+  return fourieraniso;
+}
+
+/*----------------------------------------------------------------------*
+ *----------------------------------------------------------------------*/
+Mat::FourierAniso::FourierAniso() : params_(nullptr) {}
+
+/*----------------------------------------------------------------------*
+ *----------------------------------------------------------------------*/
+Mat::FourierAniso::FourierAniso(Mat::PAR::FourierAniso* params) : params_(params) {}
+
+/*----------------------------------------------------------------------*
+ *----------------------------------------------------------------------*/
+void Mat::FourierAniso::pack(Core::Communication::PackBuffer& data) const
+{
+  // pack type of this instance of ParObject
+  int type = unique_par_object_id();
+  add_to_pack(data, type);
+
+  // matid
+  int matid = -1;
+  if (params_ != nullptr) matid = params_->id();  // in case we are in post-process mode
```

Comment:
I don't think that pack is called in post-process mode.

```suggestion
  int matid = params_->id();
```

Reply:
Removed the if statement.